tests/acl/test_acl_dscp_change.py

```diff
@@ -0,0 +1,657 @@
+    duthost.shell("redis-cli -n 4 del 'ACL_TABLE|OVERLAY_MARK_META_TESTV6' ")
+
+
+def create_acl_Rule(duthost, tableName, ruleNumber, priority, dscpAction, dstIp, qualifer=None):
+    # {   "ACL_RULE": {
+    #         "{{ tableName }}|RULE{{ruleNumber}}": {
+    #             "PRIORITY": {{priority}},
+    #             "DSCP_ACTION": {{dscpAction}},
+    #             {{qualifer}}
+    #             "DST_IP": {{dstIp}}
+
+    #         }
+    #     }
+    # }
+    if tableName == TABLE_V4:
+        qualifer = '"DST_IP":"{}"'.format(dstIp) + (',' + qualifer if qualifer is not None else "")
+    else:
+        qualifer = '"DST_IPV6":"{}"'.format(dstIp) + (',' + qualifer if qualifer is not None else "")
+    extra_vars = {
+        'tableName': tableName,
+        'ruleNumber': ruleNumber,
+        'priority': priority,
+        'dscpAction': dscpAction,
+        'qualifer': qualifer,
+       }
+
+    dest_path = os.path.join(TMP_DIR, ACL_RULES_FILE)
+    duthost.host.options['variable_manager'].extra_vars.update(extra_vars)
+    duthost.file(path=dest_path, state='absent')
+    duthost.template(src=os.path.join(TEMPLATES_DIR, ACL_TEMPLATE_RULE_FILE), dest=dest_path)
+    Logger.info("Creating ACL rule for %s with id %s, priority %s, dscp action %s, DST IP %s and qualifers %s",
+                tableName, ruleNumber, priority, dscpAction, dstIp, qualifer)
+
+    duthost.shell("config load -y {}".format(dest_path))
+
+
+def del_acl_rule(duthost, table, ruleId):
+    key = f'ACL_RULE|{table}|RULE{ruleId}'
+    duthost.shell(f"redis-cli -n 4 del '{key}'")
+
+
+def parse_matchfields(match_fields=None, separator=':'):
+    ipSrc = None
+    l4SrcPort = None
+    l4DstPort = None
+    ipTos = 0x84
+
+    if match_fields is not None:
+        qualifiers = match_fields.split(',')
+        for qual in qualifiers:
+            field, value = qual.split(separator)
+            if separator == ':"':
+                value = value.split('"')[0]
+            else:
+                value = value.split('"')[1]
+            if field == '"SRC_IP"':
+                ipSrc = value
+            elif field == '"L4_SRC_PORT"':
+                l4SrcPort = int(value)
+            elif field == '"L4_DST_PORT"':
+                l4DstPort = int(value)
+            elif field == '"DSCP"':
+                ipTos = int(value) * 4
+            elif field == '"SRC_IPV6"':
+                ipSrc = value
+    return ipSrc, l4SrcPort, l4DstPort, ipTos
+
+
+def create_expected_packet(setUp_vnet, duthost, encap_type, expectedDscp, inner_packet):
+    outer_ip_src = setUp_vnet['loopback_v4'] if 'in_v4' in encap_type else setUp_vnet['loopback_v6']
+    vxlan_vni = list(setUp_vnet[encap_type]['vnet_vni_map'].values())[0]
+
+    if 'v4_in_v4' == encap_type:
+        exp_pkt = testutils.simple_vxlan_packet(
+            eth_src=duthost.facts['router_mac'],
+            ip_src=outer_ip_src,
+            ip_dst="0.0.0.0",  # We don't care about the outer dest IP
```

Comment:
Please use a valid dummy IP address. We don't know if the packet will be dropped by leaf fanout if the dst_ip is 0.0.0.0